main.c

```diff
@@ -459,11 +460,51 @@ static void dbus_name_lost(GDBusConnection *connection,
 	tcmu_dbg("name lost\n");
 }
 
+static const char *dirname_default_prefix = "/lib/modules";
+
+static char *get_modules_dirpath(void)
+{
+	struct utsname u;
+	char *p;
+
+	if (uname(&u) < 0)
+		return NULL;
+
+	if (asprintf(&p, "%s/%s", dirname_default_prefix, u.release) < 0)
+		return NULL;
+
+	return p;
+}
+
 static int load_our_module(void)
 {
 	struct kmod_list *list = NULL, *itr;
 	struct kmod_ctx *ctx;
 	int ret;
+	struct stat sb;
+	char *modules_dir;
+
+	modules_dir = get_modules_dirpath();
+	if (!modules_dir) {
+		tcmu_err("get_modules_dirpath() failed\n");
+		return -1;
+	}
+
+	/* check if path entry exist */
```

Comment:
This is a pretty useless comment. It is clear we are checking for its existing. In the code here, could you add a comment why we need to do this?  Why doesn't kmod lib not work if /lib/modules does not exist? Is it a bug or design decision? Why do we care?

In the git commit message, could you add similar info and why the kmod devs do not want to handle it? If there is a upstream discussion link to it.

This way, if kmod changes or the container setup does, we can drop hacks in our code.